File: mobatech-flutter/lib/features/pharmacy/models/medicine.dart
import 'medicine_category.dart';

class Medicine {
  final int id;
  final String name;
  final String genericName;
  final double price;
  final int stock;
  final bool requiresPrescription;
  final String imageUrl;
  final MedicineCategory? category;

  Medicine({
    required this.id,
    required this.name,
    required this.genericName,
    required this.price,
    required this.stock,
    required this.requiresPrescription,
    required this.imageUrl,
    this.category,
  });

  factory Medicine.fromJson(Map<String, dynamic> json) {
    return Medicine(
      id: json['id'] as int,
      name: json['name'] as String,
      genericName: json['generic_name'] as String? ?? '',
      price: (json['price'] as num).toDouble(),
      stock: json['stock'] as int? ?? 0,
      requiresPrescription: json['requires_prescription'] as bool? ?? false,
      imageUrl: json['image_url'] as String? ?? '',
      category: json['category'] != null
          ? MedicineCategory.fromJson(json['category'] as Map<String, dynamic>)
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'generic_name': genericName,
      'price': price,
      'stock': stock,
      'requires_prescription': requiresPrescription,
      'image_url': imageUrl,
      'category': category?.toJson(),
    };
  }
}

File: mobatech-backend/models/medicine.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Medicine struct {
	ID                   uint             `json:"id" gorm:"primarykey"`
	CreatedAt            time.Time        `json:"created_at"`
	UpdatedAt            time.Time        `json:"updated_at"`
	DeletedAt            gorm.DeletedAt   `json:"deleted_at" gorm:"index"`
	CategoryID           uint             `json:"category_id"`
	Category             MedicineCategory `json:"category" gorm:"foreignKey:CategoryID"`
	Name                 string           `json:"name"`
	GenericName          string           `json:"generic_name"`
	Description          string           `json:"description" gorm:"type:text"`
	Dosage               string           `json:"dosage"`
	Unit                 string           `json:"unit"`
	Price                float64          `json:"price"`
	Stock                int              `json:"stock"`
	RequiresPrescription bool             `json:"requires_prescription"`
	Manufacturer         string           `json:"manufacturer"`
	ImageURL             string           `json:"image_url"`
}
